lint-staged.config.js

```diff
@@ -9,23 +9,18 @@ const tsConfigPath = './tsconfig.lint-staged.json'
 const tsConfig = (files) => `
 {
   "extends": "./tsconfig.json",
-  "include": [
-    "./src/**/*.d.ts",
-    "./types",
-    ${files.join(',\n')}
-  ],
   "compilerOptions": {
     "noEmit": true
   }
 }    
 `
 
 module.exports = {
-  '**/*.{js,jsx,ts,tsx}': async (files) => {
-    const ignored = await Promise.all(files.map(async (file) => eslint.isPathIgnored(file)))
-    const filtered = files.filter((_, index) => !ignored[index]).map((path) => `"${path}"`)
-    
-    await writeFileAsync(tsConfigPath, tsConfig(filtered))
-    return [`eslint --no-ignore --max-warnings=0 ${filtered.join(' ')} --fix`, `tsc --p ${tsConfigPath}`]
-  },
+    'src/**/*.{js,jsx,ts,tsx}': async (files) => {
```

Comment:
have you tested does lint staged works now  ? especially when you commit file referencing to some .d.ts from source or to some file from /types in imports

Reply:
I guess it works.
last 2 commits were around a file using package @gooddollar/goodprotocol where we have additional typings definined in <root>/types

Or should it be tested in a different way?

Reply:
@L03TJ3 yes pls test it explicitly
- change some js,jsx,ts,tsx files from src
- also change some file matching those wildcard '/types/*/*d.ts', '/src/**/*.test.ts*', '/**/*.*js', '*.config.ts'
- do a commit from cmdline
- check which files been added to eslint command

Reply:
Tested, expected files to lint against